src/com/facebook/buck/jvm/kotlin/KotlinLibraryDescription.java

```diff
@@ -162,8 +162,23 @@ public BuildRule createBuildRule(
     }
   }
 
+  public enum AnnotationProcessingTool {
+    /**
+     * Default tool for Kotlin modules. Allows to run Java annotation processors against
+     * Kotlin sources while backporting it for Java sources too.
+     */
+    KAPT,
+
+    /**
+     * Works only against Java sources, Kotlin sources won't have access to generated
+     * classes at compile time.
+     */
+    JAVAC,
+  }
+
   public interface CoreArg extends JavaLibraryDescription.CoreArg {
     ImmutableList<String> getExtraKotlincArguments();
+    Optional<AnnotationProcessingTool> getAnnotationProcessingTool();
```

Comment:
Be sure to document this in the rules. E.g. `docs/rule/android_library.soy`

Reply:
added 👍